theano/gof/op.py

```diff
@@ -733,9 +733,20 @@ def make_thunk(self, node, storage_map, compute_map, no_recycling):
         node_output_storage = [storage_map[r] for r in node.outputs]
         node_input_compute = [compute_map[r] for r in node.inputs]
         node_output_compute = [compute_map[r] for r in node.outputs]
-        #logger.debug('Compiling node %i of graph' % node_idx)
+
         if self._op_use_c_code:
             try:
+                # float16 get special treatment since running
+                # unprepared C code will get bad results.
+                if not getattr(self, '_f16_ok', False):
+                    def is_f16(t):
+                        return getattr(t, 'dtype', '') == 'float16'
+
+                    if (any(is_f16(i.type) for i in node.inputs) or
+                            any(is_f16(o.type) for o in node.outputs)):
+                        print ("Disabling C code for %s due to unsupported "
```

Comment:
Maybe a warning would help only notifying once per Op, but I want to make sure this is seen by the user.  It has performance implications.

Also, since you will only ever see this message if using float16, I think it can stay as a print for now.  We can change it to a warning after we have most of the work done.